src/core/patterns.js

```diff
@@ -59,3 +59,61 @@ foam.CLASS({
     }
   ]
 });
+
+/** Causes an class to pool its instances. create() will pull from the pool,
+ and destroy() will return instances to the pool. Object pools can be found
+ in <code>foam.__objectPools__</code>. */
+foam.CLASS({
+  package: 'foam.pattern',
+  name: 'Pooled',
+  axioms: [ foam.pattern.Singleton.create() ],
+
+  methods: [
+    function installInClass(cls) {
+      // Keeping the object pools in an accessible location allows them
+      // to be cleared out.
+      if ( ! foam.__objectPools__ ) { foam.__objectPools__ = {}; }
+      var pool = foam.__objectPools__[cls];
+      if ( ! pool ) {
+        pool = foam.__objectPools__[cls] = [];
+      }
+
+      var oldCreate = cls.create;
+      cls.create = function(args, X) {
+        var nu;
+        // Pull from the pool, run the usual init process that .create() would
+        // do. TODO: Alter create to accept the base object so we don't duplicate
+        // this init code?
+        if ( pool.length ) {
+          nu = pool.splice(-1, 1)[0];
+          nu.destroyed = false;
+          nu.initArgs(args, X);
+          nu.init && nu.init();
+        } else {
+          nu = oldCreate.apply(this, arguments);
+        }
+        return nu;
+      }
+
+      var oldDestroy = cls.prototype.destroy;
```

Comment:
What about doing cls.installAxiom(foam.core.Method.create({ name: 'destroy', code: function() { ... this.SUPER(); } }));  Then it will show in .describe() and IDEs nicely.